File: native/mixdog-graph/tests/fixtures/calls/swift/sample.swift
// hidden()
let quoted = "hidden()"

func inner() {}
func leaf(_ x: Int) -> Int { x }
func nest(_ x: Int) {}
func helper() {}
func seed() {}
func plain() {}

class Chain {
  func b() -> Chain { self }
  func c() -> Chain { self }
}

plain()

func run(_ a: Chain) {
  inner()
  nest(leaf(1))
  a.b().c()
}

class Widget {
  func ping() {}
  func act() {
    helper()
    self.ping()
  }
}

let mark = "μ"; seed()
_ = Widget()

File: native/mixdog-graph/tests/fixtures/calls/scala/sample.scala
// hidden()
val quoted = "hidden()"

def inner(): Unit = ()
def leaf(x: Int): Int = x
def nest(x: Int): Unit = ()
def helper(): Unit = ()
def seed(): Unit = ()
def plain(): Int = 0

class Chain {
  def b(): Chain = this
  def c(): Chain = this
}

plain()

def run(a: Chain): Unit = {
  inner()
  nest(leaf(1))
  a.b().c()
}

class Widget {
  def ping(): Unit = ()
  def act(): Unit = {
    helper()
    this.ping()
  }
}

val mark = "μ"; seed()
new Widget()
